pkg/cluster/nodes/node.go

```diff
@@ -403,42 +403,46 @@ func (n *Node) WriteFile(dest, content string) error {
 	return nil
 }
 
-var proxyEnvs = []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY"}
-
 // SetProxy configures proxy settings for the node
 //
 // Currently it only creates systemd drop-in for Docker daemon
 // as described in Docker documentation: https://docs.docker.com/config/daemon/systemd/#http-proxy
 //
 // See also: NeedProxy
-func (n *Node) SetProxy() error {
+func (n *Node) SetProxy(proxyEnvs map[string]string) error {
 	// configure Docker daemon to use proxy
 	proxies := ""
-	for _, name := range proxyEnvs {
-		val := os.Getenv(name)
-		if val != "" {
-			proxies += fmt.Sprintf("\"%s=%s\" ", name, val)
-		}
+	for key, val := range proxyEnvs {
+		proxies += fmt.Sprintf("\"%s=%s\" ", key, val)
 	}
 
-	if proxies != "" {
-		err := n.WriteFile("/etc/systemd/system/docker.service.d/http-proxy.conf",
-			"[Service]\nEnvironment="+proxies)
-		if err != nil {
-			errors.Wrap(err, "failed to create http-proxy drop-in")
-		}
+	err := n.WriteFile("/etc/systemd/system/docker.service.d/http-proxy.conf",
+		"[Service]\nEnvironment="+proxies)
+	if err != nil {
+		errors.Wrap(err, "failed to create http-proxy drop-in")
 	}
 
 	return nil
 }
 
-// NeedProxy returns true if the host environment appears to have proxy settings
+// NeedProxy returns a struct with the host environment appears to have proxy settings
```

Comment:
This should be renamed if we're going to refactor it this way.

I also think the map thing should be an internal detail, we may need proxy options in the future that are not environment variable based, in which case making the map public will make fixing that a breaking change.

Reply:
```go
// NeedProxy returns true if the host environment appears to have proxy settings
func NeedProxy() bool {
    details := getProxyDetails()
    return len(details) > 0
}

func (n *Node) SetProxy() {
    details := getProxyDetails()
    // now set the the proxy settings
}

func getProxyDetails() map[string]string {

}

```